Index: linux-3.10.49/arch/mips/include/asm/mach-ath79/ath79.h
===================================================================
--- linux-3.10.49.orig/arch/mips/include/asm/mach-ath79/ath79.h	2015-01-17 10:30:12.003918717 +0000
+++ linux-3.10.49/arch/mips/include/asm/mach-ath79/ath79.h	2015-01-17 10:39:21.171847045 +0000
@@ -144,6 +144,7 @@
 static inline void ath79_reset_wr(unsigned reg, u32 val)
 {
 	__raw_writel(val, ath79_reset_base + reg);
+       (void) __raw_readl(ath79_reset_base + reg); /* flush */
 }
 
 static inline u32 ath79_reset_rr(unsigned reg)
